Source/WebGPU/WGSL/AST/ASTShaderModule.h

```diff
@@ -40,8 +40,9 @@ class ShaderModule final : Node {
     WTF_MAKE_FAST_ALLOCATED;
 
 public:
-    ShaderModule(SourceSpan span, GlobalDirective::List&& directives, Decl::List&& decls)
+    ShaderModule(SourceSpan span, const String& source, GlobalDirective::List&& directives, Decl::List&& decls)
```

Comment:
Should `ShaderModule` be an AST node or the root container for all data that needs to be kept alive for the duration of compiling a WGSL program? (This was an idea I was toying with before @tadeuzagallo joined the project to keep the allocation for the resolved type hierarchy on the ShaderModule)

Reply:
I think it's a good idea to make it into a container